packages/components/src/presentation-components/tree/controlled/UsePresentationTreeState.ts

```diff
@@ -212,13 +227,17 @@ function useTreeState(props: TreeStateProps) {
     };
   }, [props]);
 
-  const onReload = useCallback((reloadedTree: ReloadedTree) => {
-    prevStateRef.current?.dataProvider.dispose();
+  const onReload = useCallback(
+    (reloadedTree: ReloadedTree) => {
+      prevStateRef.current?.dataProvider.dispose();
 
-    const { modelSource, dataProvider } = reloadedTree;
-    const nodeLoader = new PagedTreeNodeLoader(dataProvider, modelSource, dataProvider.pagingSize!);
-    setState({ dataProvider, nodeLoader });
-  }, []);
+      const { modelSource, dataProvider } = reloadedTree;
+      const pagedLoader = new PagedTreeNodeLoader(dataProvider, modelSource, dataProvider.pagingSize!);
+      const nodeLoader = props.onNodeLoaded ? new ReportingTreeNodeLoader(pagedLoader, props.onNodeLoaded) : pagedLoader;
+      setState({ dataProvider, nodeLoader });
+    },
+    [props.onNodeLoaded],
```

Comment:
Can you add `onNodeLoaded` function into `useRef` so that `usePresentationTreeState` confumers don't need to memoize this function?

Reply:
Added into `useRef`.